### PR TITLE
Projectile fixes

### LINKED ISSUE #1006: Throwing an egg to the ground fails
**Describe the bug**
When throwing an egg (chicken-produced) to the ground, you lose the egg, the screen flashes light grey very briefly, and nothing else happens.

**To Reproduce**
1. Throw an egg to the ground.

**Expected behavior**
Egg breaking animation. Egg throwing sound. A chicken may appear or not appear.

**Screenshots / Videos**
Video showing the problem: [https://youtu.be/QK9gVQxHc_o](https://youtu.be/QK9gVQxHc_o)

**Server Version**
PaperMC build 111. Geyser-Spigot plugin.

**Geyser Version**
Geyser-Spigot build 280 (git-master-b103d86)

**Minecraft: Bedrock Edition Version**
1.16.10.02

**Additional Context**
If a dump is needed, I'll be happy to provide one. Just drop me a line here.

### PR DESCRIPTION
**Pull request by davchoo**
Fixes #1006 and other thrown projectiles. Needs GeyserMC/mappings#39

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntityStatusTranslator.java [hunk 1]
```diff
@@ -88,6 +90,22 @@ public void translate(ServerEntityStatusPacket packet, GeyserSession session) {
                 break;
             case LIVING_DEATH:
                 entityEventPacket.setType(EntityEventType.DEATH);
+                if (entity.getEntityType() == EntityType.THROWN_EGG) {
+                    for (int i = 0; i < 6; i++) {
+                        LevelEventPacket particlePacket = new LevelEventPacket();
+                        particlePacket.setType(LevelEventType.PARTICLE_GENERIC_SPAWN);
+                        particlePacket.setData(22544397);
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntityStatusTranslator.java:97]: Is there any sort of constant where this value is stored? Not sure how I feel with this magic number just sitting here.

**Review by RednedEpic — DISMISSED** [review]

File: connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java [hunk 2]
```diff
@@ -26,11 +26,65 @@
...
+            session.sendUpstreamPacket(particlePacket);
+        }
+        return super.despawnEntity(session);
+    }
+
+    @Override
+    public void moveRelative(GeyserSession session, double relX, double relY, double relZ, Vector3f rotation, boolean isOnGround) {
+        setPosition(lastPosition);
+        super.moveRelative(session, relX, relY, relZ, rotation, isOnGround);
+        lastPosition = getPosition();
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java:82]: You can just do `lastPosition = this.position` here.

File: connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java [hunk 3]
```diff
@@ -26,11 +26,65 @@
...
+            particlePacket.setType(LevelEventType.PARTICLE_TELEPORT);
+            particlePacket.setPosition(getPosition());
+            session.sendUpstreamPacket(particlePacket);
+        }
+        return super.despawnEntity(session);
+    }
+
+    @Override
+    public void moveRelative(GeyserSession session, double relX, double relY, double relZ, Vector3f rotation, boolean isOnGround) {
+        setPosition(lastPosition);
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java:80]: Not sure why you have to do this? `moveRelative` sets the position, just as `this.position = ...`

File: connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java [hunk 4]
```diff
@@ -26,11 +26,65 @@
...
+        super.moveRelative(session, relX, relY, relZ, rotation, isOnGround);
+        lastPosition = getPosition();
+    }
+
+    @Override
+    public void moveAbsolute(GeyserSession session, Vector3f position, Vector3f rotation, boolean isOnGround, boolean teleported) {
+        super.moveAbsolute(session, position, rotation, isOnGround, teleported);
+        lastPosition = getPosition();
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java:88]: See comment in `moveRelative`.

**Review by DoctorMacc — COMMENTED** [review]
Functionally, this PR works great and it's so awesome to see items moving so nicely!

File: connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntityStatusTranslator.java [hunk 5]
```diff
@@ -91,18 +98,21 @@ public void translate(ServerEntityStatusPacket packet, GeyserSession session) {
             case LIVING_DEATH:
                 entityEventPacket.setType(EntityEventType.DEATH);
                 if (entity.getEntityType() == EntityType.THROWN_EGG) {
+                    ItemEntry eggItemJava = ItemRegistry.getItemEntry("minecraft:egg");
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntityStatusTranslator.java:101]: Wow, I did not realize that's what the number was! Nice catch.

As this number isn't going to change on us, perhaps make it a static constant in `ItemRegistry`? (Match what we do there with the other items)

**Comment by DoctorMacc** [comment]
I merged your mappings changes into master; bump the submodule here and it's good to go in my eyes.

**Review by DoctorMacc — APPROVED** [review]

**Comment by zimzaza4** [comment]
Fireball still has problems

**Pull request merged**